### PR TITLE
WIP booking confirmation flow

### PR DESCRIPTION
**Pull request by neobay991**
requires two new columns in the listings table
- available (type: bool)
- requested (type: bool)

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -27,5 +27,25 @@ class BnB < Sinatra::Base
 
   get '/users/new' do
     erb(:users_new)
-  end 
+  end
+
+  get '/users/my-listings' do
```
Comment (heuperman) [inline app.rb]: restful => 'users/#{user.id}/listings'

File: app.rb [hunk 2]
```diff
@@ -27,5 +27,25 @@ class BnB < Sinatra::Base
...
+    @listings_pending = Listing.all_open_requests
+    @listings_not_pending = Listing.all_closed_requests
```
Comment (heuperman) [inline app.rb]: Change to @requests_pending and @requests_confirmed?

File: app.rb [hunk 3]
```diff
@@ -27,5 +27,25 @@ class BnB < Sinatra::Base
...
+    erb(:users_my_listings)
+  end
+
+  post '/users/my-listings' do
+    Listing.book_listing(params['id'])
+    redirect '/users/my-listings'
```
Comment (heuperman) [inline app.rb]: Change both post and redirect 'users/:user.id/listings'

File: lib/listing.rb [hunk 4]
```diff
@@ -40,6 +40,43 @@ def self.create_new_listing(result)
...
+  # this methods manages the request for a booking
+  def self.request_booking(id)
+    # puts id
+    connection = connect_database
+    connection.exec("UPDATE listings SET requested = 'true' WHERE id = '#{id}'")
+    # puts "listing request updated"
+  end
+
+  # this methods manages the booking
+  def self.book_listing(id)
```
Comment (heuperman) [inline lib/listing.rb]: Rename to confirm_booking?

File: views/users_listing_request_confirmation.erb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+Thanks! Your booking request has been sent.
+
+Return to the booking <a href="/listings"listing page</a>.
```
Comment (heuperman) [inline views/users_listing_request_confirmation.erb]: missing a > after "/listings"

**Comment by neobay991** [comment]
thanks, made updates :-)

File: app.rb (hunk 1 above)
Comment (neobay991) [reply app.rb]: 👍  - once we hook up user

**Pull request merged**